src/python_routing_v02/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -1155,19 +1155,32 @@ cpdef object compute_network_structured(
                 MC_Reach(segment_objects, array('l',upstream_ids))
                 )
             
+    # replace initial conditions with gage observations, wherever available
+    cdef int gages_size = usgs_positions_list.shape[0]
+    cdef int gage_i, usgs_position_i
+    cdef int gage_maxtimestep = usgs_values.shape[1]
+
+    if gages_size:
+        for gage_i in range(gages_size):
+            usgs_position_i = usgs_positions_list[gage_i]
+            flowveldepth_nd[usgs_position_i, 0, 0] = usgs_values[gage_i, 0]
+            # TODO: handle the instance where there are no values, only gage positions
+    
     cdef np.ndarray fill_index_mask = np.ones_like(data_idx, dtype=bool)
     cdef Py_ssize_t fill_index
     cdef long upstream_tw_id
     cdef dict tmp
     cdef int idx
     cdef float val
 
-    for upstream_tw_id in upstream_results:
+    for upstream_tw_id in upstream_results:  
         tmp = upstream_results[upstream_tw_id]
         fill_index = tmp["position_index"]
         fill_index_mask[fill_index] = False
         for idx, val in enumerate(tmp["results"]):
-            flowveldepth_nd[fill_index, np.floor(idx//3).astype('int')+1, idx%3] = val
+            flowveldepth_nd[fill_index, (idx//3) + 1, idx%3] = val
+            flowveldepth_nd[fill_index, 0, 0] = init_array[fill_index, 0] # initial flow condition
+            flowveldepth_nd[fill_index, 0, 2] = init_array[fill_index, 2] # initial depth condition
```

Comment:
Here, I am populating the initial conditions for off-network upstream segments in the by-subnetwork parallel schemes. Without this change, initial conditions of off-network upstream segments are set to zero.

Reply:
- [ ] consider updating "magic" number '3' with a var `nvars` or perhaps `qvd_ts_w` to be consistent with a recent change in the `V02-caching` compute kernel.